### DISCUSSION TITLE
MaterialX Support

### DISCUSSION THREAD

**Discussion by fantasy-fish** [mitsuba-renderer/mitsuba3#704] [Feature requests]
Coming from the ASWF [materialX ](https://materialx.org/index.html) group. We are thinking that image-based material generation is a very promising area. To do that, we are thinking integration of Mitsuba3 into MaterialX or the other way around. My initial idea would be to add diff rendering support for only a few attributes, such as diffuse map, reflectance values, and so on. But in order to do that, we need to know the data for lights and camera, which is usually out of the scope of MaterialX. Can someone help me find a feasible solution for this?

**Comment by njroussel** [comment]
Hi @fantasy-fish 

> But in order to do that, we need to know the data for lights and camera, which is usually out of the scope of MaterialX.

I'm a bit confused by this statement. The lights and cameras are only used for the reconstruction/optimization, right?
Ideally, your virtual scene would perfectly match the setting in which you acquired the input images.

**Reply by njroussel** [reply]
Let me add to this. We don't have any plan to support MaterialX ourselves, it's out of the scope of this project. I still encourage you to try this - I don't see why it wouldn't work - but we simply do not need nor have the resources to work and maintain such a feature.
